src/main/java/org/carlspring/cloud/storage/s3fs/S3FileChannel.java

```diff
@@ -222,15 +241,14 @@ protected void sync()
     {
         try (InputStream stream = new BufferedInputStream(Files.newInputStream(tempFile)))
         {
-            ObjectMetadata metadata = new ObjectMetadata();
-            metadata.setContentLength(Files.size(tempFile));
-            metadata.setContentType(new Tika().detect(stream, path.getFileName().toString()));
-
-            String bucket = path.getFileStore().name();
-            String key = path.getKey();
-
-            path.getFileSystem().getClient().putObject(bucket, key, stream, metadata);
+            final PutObjectRequest.Builder builder = PutObjectRequest.builder();
```

Comment:
If the temp file is larger than 5 GB then instead of a put you will need a multi-part upload.

Reply:
Thanks for flagging this!

In fact, while digging through some of the pull requests of the [Upplication/Amazon-S3-FileSystem-NIO2](https://github.com/Upplication/Amazon-S3-FileSystem-NIO2/) project, I saw that somebody actually already implemented this in one of the forks.

I think that we should add a follow-up task for this and look into it as well. @ptirador , would you like to create one? :)

Reply:
@carlspring : Follow-up task created: https://github.com/carlspring/s3fs-nio/issues/95.

Reply:
Excellent! Thank you! :)